stlab/concurrency/set_current_thread_name.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef STLAB_SET_CURRENT_THREAD_NAME_HPP
+#define STLAB_SET_CURRENT_THREAD_NAME_HPP
+
+/**************************************************************************************************/
+#if defined(__APPLE__)
+
+#include <pthread.h>
+
+/**************************************************************************************************/
+#elif defined(_MSC_VER)
+
+#include <cstring>
+#include <string>
+
+#include <windows.h>
+
+#include <processthreadsapi.h>
+#include <stringapiset.h>
+
+/**************************************************************************************************/
+#elif defined(__EMSCRIPTEN__) && defined(__EMSCRIPTEN_PTHREADS__)
+
+#include <pthread.h>
+
+#include <emscripten/threading.h>
+
+/**************************************************************************************************/
+#elif defined(__has_include) && __has_include(<pthread.h>)
+
+#include <pthread.h>
+
+/**************************************************************************************************/
+#endif
+
+/**************************************************************************************************/
+
+namespace stlab {
+
+/**************************************************************************************************/
+#if defined(__APPLE__)
+
+inline void set_current_thread_name(const char* name) { pthread_setname_np(name); }
+
+/**************************************************************************************************/
+#elif defined(_MSC_VER)
+
+inline void set_current_thread_name(const char* name) {
+    /* Should string->wstring be split out to a utility? */
```

Comment:
At the moment I don't think so, because this is the very only place to handle anything with strings explicitly.